src/internal/client/storage.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"context"
+	"io"
+	"io/fs"
+
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/pachyderm/pachyderm/v2/src/storage"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+)
+
+// FileSystemToFileset creates a new fileset containing fileSystem, returning the fileset ID.
+func (c APIClient) FileSystemToFileset(ctx context.Context, fileSystem fs.FS) (string, error) {
+	cc, err := c.FilesetClient.CreateFileset(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "could not create fileset")
+	}
+	if err := fs.WalkDir(fileSystem, ".", func(path string, d fs.DirEntry, err error) error {
```

Comment:
Just to confirm, are entries returned lexicographically with directories?

So if the structure was:
```
/
a/
b
```
and the structure of a/ was
```
a/c
a/d
```
would we get:
```
a/c
a/d
b
```
in the output?

Reply:
[Yup](https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/testing/fstest/mapfs.go;l=103-105)